src/shims/fs.rs

```diff
@@ -139,9 +142,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
-        this.remove_handle_and(fd, |handle, this| {
+        if let Some(handle) = this.machine.file_handler.handles.remove(&fd) {
             this.try_unwrap_io_result(handle.file.sync_all().map(|_| 0i32))
```

Comment:
Please add a comment explaining why we do this. I expected a `drop(handle)` and had to go look up some docs to understand this.
Also, the drop still happens, right? Might be good to make it explicit.

Reply:
am I thinking naively that the handle is dropped after it goes out of scope?

Reply:
You are right, but I find it rather odd for a "close" function to do its most important operation -- closing the file -- entirely implicitly, without any code to point at.
It is sometimes a good idea to explicitly drop stuff, e.g. when dropping it has some important side-effects -- like closing a file handle.